core/trino-main/src/test/java/io/trino/execution/TestSqlTaskManager.java

```diff
@@ -265,6 +266,8 @@ public void testFailStuckSplitTasks()
         taskExecutor.enqueueSplits(taskHandle, false, ImmutableList.of(mockSplitRunner));
         taskExecutor.start();
 
+        // Wait for taskExecutor to process splits
+        Thread.sleep(100);
```

Comment:
This most likely will be flaky too (e.g. when node becomes slow). Can we somehow signal or observe when splits processing is finished?

Reply:
fixed